Add flows (#31)
axiomhq/docs: transfer-data/flows.mdx

---
title: 'Flows'
description: 'This page explains how to set up a flow to transfer and transform data from an Axiom dataset to a destination.'
---

import Prerequisites from "/snippets/minimal-prerequisites.mdx"

<Note>
Flow is currently in preview. To try it out, [sign up for a free preview](https://forms.axiom.co/flow).
</Note>

> A flow is a way to transfer and transform data from an Axiom dataset to a destination that you choose. This page explains how to set up a flow.

Reviewer: ```suggestion
A flow is a way to transform and transfer data from an Axiom dataset to a destination that you choose. This page explains how to set up a flow.
```

As above

<Prerequisites />
{/* list separator */}
- Set up a destination. For more information, see [Destinations](/transfer-data/destinations).

## Set up a flow

To set up a flow:
1. Click the **Flows** tab. Axiom displays the list of flows you have created.
2. In the top right, click **New flow**.

> 3. In the **Source** section, specify the source dataset and the transformation in an APL query. For example, the following APL query selects events from the `cloudflare-logpush` dataset and transforms the data by deleting and adding a field.

Reviewer: ```suggestion
3. In the **Source** section, specify the source dataset and the transformation in an APL query. For example, the following APL query selects events from the `cloudflare-logpush` dataset and transforms the data by removing a set of fields, and enriching with a new one.
```

To be precise, `['@app']*` line is removing several fields, so this tries to communicate that

    ```kusto

>     // Transfer high-value app events to a costly SIEM
>     // while retaining the complete record of all events in Axiom.

Reviewer: ```suggestion
```

I would remove this tbh. I don’t think it makes so much sense in this context. Better described in the numbered bullet above if wanting to include imo.

    ['cloudflare-logpush']
    | where QueryName == "app.axiom.co."
    // Reduce events by dropping unimportant field
    | project-away ['@app']*
    // Enrich events with additional context
    | extend ['@origin'] = "_axiom"
    ```

    <Note>

>     If you only specify the name of the dataset in the query, Axiom transfers all the raw data to the destination.

Reviewer: ```suggestion
    If you only specify the name of the dataset in the query, Axiom transfers all the raw data to the destination.
```
```suggestion
    If you only specify the name of the dataset in the query, Axiom transfers all events to the destination.
```

Prefer events?

Reviewer 2: Two things: 1) if possible try to discourage queries without predicate; 2) make a point of the actual APL subset acceptable for Axiom Flow.

Author: @ucirello Thanks. What is the acceptable subset of APL in this case?

    </Note>

> 4. Click **Preview** to check whether the query you specified transforms your data as you want it. The **Input event** section displays the original data stored in Axiom. The **Output event** section displays the transformed data that Axiom sends to the destination. The raw data in Axiom isn’t affected by the transformation.

Reviewer: > The raw data in Axiom isn’t affected by the transformation.

I like this detail 👍

Author: Thank you!

5. In the **Destinations** section, click **Add destination**, and then select the destination where you want to transfer the transformed data. To add several destinations, click **+** in the top right corner.
6. In the top right, click **Create**.

> GIF

Reviewer: TODO!

> ## Replay a flow

Reviewer: I would like us to include a note that there are three kinds of flow: one-time, scheduled, and continuous, and that the preview will support one-time flows to begin with. This is critical for customers to understand.
